frontend/src/metabase/common/hooks/use-dashboard-query/use-dashboard-query.unit.spec.tsx

```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import LoadingAndErrorWrapper from "metabase/components/LoadingAndErrorWrapper";
+import { createMockDashboard } from "metabase-types/api/mocks";
+import { setupDashboardEndpoints } from "__support__/server-mocks";
+import {
+  renderWithProviders,
+  screen,
+  waitForElementToBeRemoved,
+} from "__support__/ui";
+import { useDashboardQuery } from "./use-dashboard-query";
+
+const TEST_DASHBOARD = createMockDashboard();
+
+const TestComponent = () => {
+  const { data, isLoading, error } = useDashboardQuery({
+    id: TEST_DASHBOARD.id,
+  });
+
+  if (isLoading || error || !data) {
+    return <LoadingAndErrorWrapper loading={isLoading} error={error} />;
```

Comment:
LoadingAndErrorWrapper is meant to wrap a component that consumes data but here it is used in a different way. Probably, we don't need it here and a simple chain of `if (error) return <div>error</div>` would work fine, and also it should allow adding the test case for the errored case